### DISCUSSION TITLE
Nicer image optimization names

### DISCUSSION THREAD

**Discussion by MrAmericanMike** [withastro/roadmap#726] [Proposal]
### Body

# Summary

Main topic was started here https://github.com/withastro/docs/issues/4990 but I was recommended to post it on the roadmap.

The main idea is to be able to customize or have more control over the generated names for optimized images.
As mentioned in the documentation:

```
Keep in mind that during development, Astro uses a src/ path, but upon building, it generates hashed paths like /_astro/cat.a6737dd3.png.
```

The hash `a6737dd3` looks just horrible and doesn't tell anything. My idea is to be able to have that, become something more like. `cat-300x200.jpg` 

Also related to this, I noticed that images in the `pages/some_route/image.jpg` end up in the `_astro` folder at build time, and even the files in `assets/some_route/image.jpg` also end up in `_astro` at build time. It would be great if Astro could do something to keep the folder structure, or give us control over that.

# Background & Motivation

Not sure what the "Background & Motivation" means. So I'll just say, I'm suggesting this, to make the output files be more "user friendly" to read, not only for visitors to the site, but for anyone looking at the build project and seen those files and been able to tell at a glance what each one represents. (If I'm not mistaken Wordpress does something like this)

# Goals

- Control over how Astro "hashes" images
- Control over in what directory the images should end up

# Example

Not applicable.

**Comment by ADTC** [comment]
We need this due to client's legacy system limitations. Is it even possible? I don't understand why Astro is opinionated about this, and not configurable. Surely there is a way to have the cake and eat it too.
